wordpress-mobile/WordPress-Android
# [Jetpack Content Migration Flow] Refactor login resolver as orchestrator

# Description

This PR continues from #17398 by refactoring the login resolver as an orchestrator. With the unified entity provider, the entity-specific logic is extracted to a helper class for the provider-side. The resolver-side is simplified to a single inlined query to obtain the data, leaving the consumption of that data to the orchestrator.

This begins the process and serves as a first step of unifying the various providers, and reducing the boilerplate needed to retrieve specified data from the provider. It also renames the existing shared login resolver class as an orchestrator to better describe its responsibility.

To test:

Follow the testing steps from: https://github.com/wordpress-mobile/WordPress-Android/pull/17398, the behavior should not change.

## Regression Notes
1. Potential unintended areas of impact
No. Existing implementations are changed, but are behind a feature flag.

9. What I did to test those areas of impact (or what existing automated tests I relied on)
Existing implementations have automated tests, and manual testing was also done to ensure things are still working as desired.

10. What automated tests I added (or what prevented me from doing so)
Unit tests were added for these implementations.

PR submission checklist:
